src/TerminalFighter/Settings/Settings.cpp

```diff
@@ -1,44 +1,232 @@
-#include <yaml-cpp/yaml.h>
 #include <glog/logging.h>
 
 #include "Settings.h"
 
-namespace {
-    const std::string VIDEO_SETTINGS_FILE = "config/video_settings.yml";
-    const std::string ASSETS_PATHS_FILE = "config/asset_paths.yml";
+// Internal helper func
+std::string vec_to_str(std::vector<std::string> vec) {
+    std::string result = "{";
+
+    for (auto str : vec) {
+        result += str + ",";
+    }
+
+    return result += "}";
 }
 
-Settings::Settings() {
+Settings::Settings(
+    std::string video_settings_file,
+    std::string asset_paths_file)
+    : video_settings_(SettingsSection::VIDEO_SETTINGS, video_settings_file)
+    , asset_paths_(SettingsSection::ASSET_PATHS, asset_paths_file) {
+    if (!reload_all_settings()) {
+        LOG(FATAL) << "Failed to initialize settings";
+    }
 }
 
 // Used for initialization as well as forced reload of settings
-bool Settings::reload_settings() {
+bool Settings::reload_all_settings() {
     LOG(INFO) << "Reloading all settings";
     int errors = 0;
 
+    errors += reload_setting(SettingsSection::VIDEO_SETTINGS);
+    errors += reload_setting(SettingsSection::ASSET_PATHS);
+
+    return !errors;
+}
+
+bool Settings::load_str(
+    SettingsSection section,
+    std::vector<std::string> keys,
+    std::string& value) const {
```

Comment:
So these `load_*` functions seems pretty solid. However seems like there's lots of repeat code.
Why not - for the scalar functions - use templates <T> :3? 

That way I can use it like:
```
int screen_height;
load_setting<int>(SettingsSection::VIDEO_SETTINGS,{"window"},{"height"}, screen_height);
```

Reply:
Discussed impossible